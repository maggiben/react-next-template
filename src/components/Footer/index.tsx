import { useEffect, useState } from 'react';
import { TextBody, Divider, SearchIcon, UserIcon, Heading, IconButton, Grid, GridItem } from '@fravega-it/bumeran-ds-fvg';
import { useTranslation } from 'react-i18next';
import pjson from '@pjson';
import React from 'react';
import styled from "styled-components";

const Container = styled.div`
  height: 2rem;
  border-width: 1px;
`;

const Right = styled.div`
  line-height: 2rem;
  padding-right: 6px;
  text-align: right;
`;

export default function Footer(): JSX.Element {
  const { t } = useTranslation();
  return (
    <Container>
      <Divider />
      <Grid>
        <GridItem xs={12} justifySelf="stretch" alignSelf="start">
            <Right>
              <TextBody size="s">{`${t('version')}: ${pjson.version}`}</TextBody>
            </Right>
        </GridItem>
      </Grid>
    </Container>
  );
}
