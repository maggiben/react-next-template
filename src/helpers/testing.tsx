import { default as baseRenderer } from 'react-test-renderer';
import { render, RenderResult } from '@testing-library/react';
import i18n from '@utils/i18n';
import { I18nextProvider } from 'react-i18next';
import { RecoilRoot } from 'recoil';
import { ThemeProvider } from 'styled-components';
import { theme, IDefaultTheme } from '@fravega-it/bumeran-ds-fvg';
import React, { ReactElement } from 'react';
import '@testing-library/jest-dom'

/**
 * Helper used to render components for testing.
 * The function appends the theme to the children.
 * */
interface IRenderer {
  create: (children: ReactElement) => RenderResult;
  baseRenderer: (children: ReactElement) => baseRenderer.ReactTestRenderer;
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme) => RenderResult;
  renderWithI18n: (children: ReactElement) => RenderResult;
}

export const renderer: IRenderer = {
  create: (children: ReactElement): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithTheme: (children: ReactElement, theme: IDefaultTheme): RenderResult =>
    render(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  baseRenderer: (children: ReactElement): baseRenderer.ReactTestRenderer =>
    baseRenderer.create(
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
  ),
  renderWithI18n: (children: ReactElement): RenderResult => 
    render(
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
      </ThemeProvider>
  ),
};