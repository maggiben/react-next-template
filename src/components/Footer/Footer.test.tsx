import React from 'react';
import { renderer } from '@helpers/testing';
import '@testing-library/jest-dom';
import Footer from './';
import { expect } from '@jest/globals';


describe('Footer', () => {
  it('renders the Footer component', () => {
    const tree = renderer.create(<Footer />);
    expect(tree).toMatchSnapshot();
  });
});
