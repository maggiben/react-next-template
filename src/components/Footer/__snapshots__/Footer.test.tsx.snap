// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders the Footer component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Footer__Container-sc-pbo7it-0 gFVabo"
      >
        <div
          class="Footer__Hr-sc-pbo7it-2 biUGVH"
        />
        <div
          class="sc-jsJBEP gDratY"
        >
          <div
            class="sc-eeDRCY ixefoN"
          >
            <div
              class="Footer__Right-sc-pbo7it-1 ILgTQ"
            >
              <svg
                class="sc-kpDqfm huIZPh"
                color="violet"
                fill="none"
                viewBox="0 0 28 28"
                width="20px"
              >
                <path
                  clip-rule="evenodd"
                  d="M15.744 7.023c.417.104.67.527.566.943L13.2 20.411a.778.778 0 0 1-1.51-.378L14.802 7.59a.778.778 0 0 1 .943-.566Zm-5.86 3.316a.778.778 0 0 1 0 1.1l-2.562 2.56 2.561 2.562a.778.778 0 1 1-1.1 1.1l-3.11-3.111a.778.778 0 0 1 0-1.1l3.11-3.111a.778.778 0 0 1 1.1 0Zm8.233 0a.778.778 0 0 1 1.1 0l3.11 3.111a.778.778 0 0 1 0 1.1l-3.11 3.111a.778.778 0 1 1-1.1-1.1L20.677 14l-2.56-2.561a.778.778 0 0 1 0-1.1Z"
                  fill-rule="evenodd"
                />
              </svg>
              <span
                class="sc-iGgWBj hTwzmc"
              >
                <span
                  class="sc-dhKdcB lcWvTs"
                >
                  Versión: 
                </span>
                1.0.2-new-ui
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Footer__Container-sc-pbo7it-0 gFVabo"
    >
      <div
        class="Footer__Hr-sc-pbo7it-2 biUGVH"
      />
      <div
        class="sc-jsJBEP gDratY"
      >
        <div
          class="sc-eeDRCY ixefoN"
        >
          <div
            class="Footer__Right-sc-pbo7it-1 ILgTQ"
          >
            <svg
              class="sc-kpDqfm huIZPh"
              color="violet"
              fill="none"
              viewBox="0 0 28 28"
              width="20px"
            >
              <path
                clip-rule="evenodd"
                d="M15.744 7.023c.417.104.67.527.566.943L13.2 20.411a.778.778 0 0 1-1.51-.378L14.802 7.59a.778.778 0 0 1 .943-.566Zm-5.86 3.316a.778.778 0 0 1 0 1.1l-2.562 2.56 2.561 2.562a.778.778 0 1 1-1.1 1.1l-3.11-3.111a.778.778 0 0 1 0-1.1l3.11-3.111a.778.778 0 0 1 1.1 0Zm8.233 0a.778.778 0 0 1 1.1 0l3.11 3.111a.778.778 0 0 1 0 1.1l-3.11 3.111a.778.778 0 1 1-1.1-1.1L20.677 14l-2.56-2.561a.778.778 0 0 1 0-1.1Z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              class="sc-iGgWBj hTwzmc"
            >
              <span
                class="sc-dhKdcB lcWvTs"
              >
                Versión: 
              </span>
              1.0.2-new-ui
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
